doc/user/content/manage/dbt/development-workflows.md

```diff
@@ -0,0 +1,358 @@
+
+## Development
+
+When you're prototyping your use case and fine-tuning the underlying data model,
+your priority is **iteration speed**. dbt has many features that can help speed
+up development, like [node selection](#node-selection) and [model preview](#preview-model-results).
+Before you start, we recommend getting familiar with how these features
+work with the `dbt-materialize` adapter to make the most of your development
+time.
+
+### Node selection
+
+By default, the `dbt-materialize` adapter drops and recreates **all** models on
+each `dbt run` invocation. This can have unintended consequences, in particular
+if you're managing sources and sinks as models in your dbt project. dbt allows
+you to selectively run specific models and exclude specific materialization
+types from each run using [node selection](https://docs.getdbt.com/reference/node-selection/syntax).
+
+#### Exclude sources and sinks
+
+{{< note >}}
+As you move towards productionizing your data model, we recommend managing
+sources and sinks [using Terraform](#other-deployment-tools) instead.
+{{</ note >}}
+
+You can manually exclude specific materialization types using the
+[`exclude` flag](https://docs.getdbt.com/reference/node-selection/exclude) in
+your dbt run invocations. To exclude sources and sinks, use:
+
+```bash
+dbt run --exclude materialized:source materialized:sink
+```
+
+##### YAML selectors
+
+Instead of manually specifying node selection on each run, you can create a
+[YAML selector](https://docs.getdbt.com/reference/node-selection/yaml-selectors)
+that makes this the default behavior when running dbt:
+
+```yaml
+# YAML selectors should be defined in a top-level file named selectors.yml
+selectors:
+  - name: exclude_sources_and_sinks
+    description: >
+      Exclude models that use source or sink materializations in the command
+      invocation.
+    default: true
+    definition:
+      union:
+        # The fqn method combined with the "*" operator selects all nodes in the
+        # dbt graph
+        - method: fqn
+          value: "*"
+        - exclude:
+            - 'config.materialized:source'
+            - 'config.materialized:sink'
+```
+
+Because `default: true` is specified, dbt will use the selector's criteria
+whenever you run an unqualified command (e.g. `dbt build`, `dbt run`). You can
+still override this default by adding selection criteria to commands, or adjust
+the value of `default` depending on the target environment. To learn more about
+using the `default` and `exclude` properties with YAML selectors, check the
+[dbt documentation](https://docs.getdbt.com/reference/node-selection/yaml-selectors).
+
+#### Run a subset of models
+
+You can run individual models, or groups of models, using the [`select` flag](https://docs.getdbt.com/reference/node-selection/syntax#how-does-selection-work)
+in your dbt run invocations:
+
+```bash
+dbt run --select "my_dbt_project_name"   # runs all models in your project
```

Comment:
Can one of the examples show the “+” the creates a model and all its dependencies? I think that will be super common

Reply:
Added.